### PR TITLE
Update docstring of ColorJitter, adjust_hue,_rgb2hsv

### LINKED ISSUE #5694: ColorJitter> adjust_hue>_rgb2hsv lack of proper documentation.
### 📚 The doc issue

Following up on #3110, it seems that lack of proper documentation on `ColorJitter`, and specifically its underlying transform `adjust_hue` which uses `_rgb2hsv` is causing issues for more and more people. https://github.com/pytorch/vision/issues/3110#issuecomment-1009688670 https://github.com/pytorch/vision/issues/3110#issuecomment-1064918900

I remember back in 2020, it took me days to track down this issue, which was causing my training pipeline to fail while a simple documentation of input requirements would save the day.

### Suggest a potential alternative/fix

The docstrings of this functions need to be updated with the information about some characteristics of the input image, and potential preprocessing that can cause these functions fail, like certain way of normalization and interpolation.

### PR DESCRIPTION
**Pull request by drbeh**
Fix #5694 

Update docstrings of `ColorJitter` and its underlying functional transforms `adjust_hue` and `_rgb2hsv` to include information about some characteristics of the input image, and potential errors caused by some preprocessing steps that can cause these functions to fail, like certain way of normalization and interpolation.

### CODE REVIEW

File: torchvision/transforms/functional_tensor.py [hunk 1]
```diff
@@ -191,6 +191,14 @@ def adjust_contrast(img: Tensor, contrast_factor: float) -> Tensor:
...
 
 def adjust_hue(img: Tensor, hue_factor: float) -> Tensor:
+    """Adjust hue of an RGB image (batch of images)
+    Args:
+        img (Tensor): a Tensor representing a 2D RGB image (CHW) or a batch of
+            2D RGB images (NCHW).
+            Note that the pixel values of the image has to be non-negative, so
+            it does not work if you normalize your image to an interval with
+            negative values like [-1, 1] before using this function.
+    """
```
Comment (datumbox) [inline torchvision/transforms/functional_tensor.py:201]: @drbeh These methods are currently considered private and that's why they are undocumented. 

The public implementations are on `functional.py` and `transforms.py`, so it's worth updating the documentation there.

**Review by datumbox — APPROVED** [review]
@drbeh LGTM, thanks!
